src/thor/dijkstras.cc

```diff
@@ -622,31 +627,20 @@ void Dijkstras::ExpandForwardMultiModal(GraphReader& graphreader,
     newcost += transition_cost;
 
     // Prohibit entering the same station as the prior.
-    if (directededge->use() == Use::kTransitConnection &&
+    // TODO: if pred was a station, we wouldn't know about it; we only set
+    // prior_stopid for platforms so far; also pull this check much further up
+    if (directededge->use() == Use::kPlatformConnection &&
         directededge->endnode() == pred.prior_stopid()) {
       continue;
     }
 
     // Test if exceeding maximum transfer walking distance
-    if (directededge->use() == Use::kTransitConnection && pred.prior_stopid().Is_Valid() &&
+    // TODO: transfer distance != walking distance! (one more label member?)
+    if (directededge->use() == Use::kPlatformConnection && pred.prior_stopid().Is_Valid() &&
         walking_distance > max_transfer_distance_) {
       continue;
     }
 
-    // Make the label in advance, we may not end up using it but we need it for the expansion decision
-    MMEdgeLabel edge_label{pred_idx,         edgeid,      directededge,    newcost,
-                           newcost.cost,     0.0f,        mode_,           path_distance,
-                           walking_distance, tripid,      prior_stop,      blockid,
-                           operator_id,      has_transit, transition_cost, restriction_idx,
-                           pred.path_id()};
-
-    // See if this is even worth expanding
-    auto maybe_expand = ShouldExpand(graphreader, edge_label, ExpansionType::multimodal);
-    if (maybe_expand == ExpansionRecommendation::prune_expansion ||
-        maybe_expand == ExpansionRecommendation::stop_expansion) {
-      continue;
-    }
```

Comment:
we already do this in the outer loop. true, we wouldn't have to add it to the queue & labels in the first place, but I did find it silly to have this check twice.